Documentation/coreos.rst

```diff
@@ -0,0 +1,356 @@
+Slack channel <https://cilium.herokuapp.com>`_. With Cilium contributors
+across the globe, there is almost always someone available to help.
+
+
+Pre-Setup / Ensure Minimum Requirements Are Met
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* Make sure you understand the various `deployment options <https://coreos.com/kubernetes/docs/latest/getting-started.html#deployment-options>`_.
+* *etcd*
+    * Cilium on CoreOS has been tested using etcd as the key-value store for Kubernetes. We recommend using at least version 3.1.0 of etcd. To set up etcd, follow `the CoreOS instructions for setting up an etcd cluster <https://coreos.com/kubernetes/docs/latest/getting-started.html#deploy-etcd-cluster>`_.
+* *kubectl*
+    * It is also required that you install ``kubectl`` version ``>= 1.6.4`` as described in the `Kubernetes Docs <https://kubernetes.io/docs/tasks/tools/install-kubectl/>`_.
+
+Setup Master Nodes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Step 1: Create TLS Assets
+--------------------------------
+Follow the `CoreOS instructions for creating TLS assets <https://coreos.com/kubernetes/docs/latest/getting-started.html#generate-kubernetes-tls-assets>`_ and `install the TLS assets needed for the master nodes <https://coreos.com/kubernetes/docs/latest/deploy-master.html#tls-assets>`_ for secure communication with the kube-apiserver.
+
+
+Step 2: Setup Kubelet
+--------------------------------
+
+On CoreOS, Kubelet (the Kubernetes agent that runs on each node, more info `here <https://kubernetes.io/docs/admin/kubelet/>`_) runs as a container. In order to securely communicate with the API server, kubelet uses the TLS assets we generated as part of Step 1.
+
+Master nodes are usually not scheduled to run workloads, so we provide the ``-register-schedulable=false`` in the example YAML excerpt below. This ensures that workloads are only scheduled on worker nodes.
+
+Since we are setting up Kubelet to use Cilium, we want to configure its networking to utilize CNI (Container Networking Interface). This ensures that each pod that is created can communicate with one another within the cluster with Cilium networking configured.
+
+* Replace ${ADVERTISE_IP} with this node's publicly routable IP.
+* Replace ${DNS_SERVICE_IP}
+* Replace ${KUBE_VERSION} with a version  ``>= 1.6.4``.
+* Cilium-specific configuration    
+
+    * Mount the CNI configuration directory you created in step 1 so Kubelet can pick up the CNI configuration from the host filesystem:
+
+         ::
+
+             --volume etc-cni,kind=host,source=/etc/cni/net.d \
+             --mount volume=etc-cni,target=/etc/cni/net.d"
+
+    * Mount the directory where CNI plugins are installed:
+
+        ::
+
+             --volume cni-bin,kind=host,source=/opt/cni/bin \
+             --mount volume=cni-bin,target=/opt/cni/bin
+
+    * Mount the BPF filesystem so that the information stored there persists across Cilium restarts:
+
+        ``ExecStartPre=/bin/mount bpffs /sys/fs/bpf -t bpf``
```

Comment:
Let's refer to here as well for more information:
http://docs.cilium.io/en/latest/admin/#mounting-the-bpf-fs